### PR TITLE
[tool_requires][best-practices] Using protobuf

### PR DESCRIPTION
**Pull request by franramirez688**
Depends on https://github.com/conan-io/examples2/pull/97

* `myaddresser`: using protobuf as `requires` and `tool_requires` at the same time (cross-building as well)

Related docs: https://github.com/conan-io/docs/pull/3222

### CODE REVIEW

**Review by memsharded — COMMENTED** [review]
Use "mylibrary`` or better something with meaning, like "myserver", as the example, with its ``test_package``, instead of using an extra ``consumer`` application package.

**Review by memsharded — APPROVED** [review]
It is looking good, but it must wait until the other PR for ``tool-requires`` is merged, then updated. Or just clean the other PR changes.

File: examples/graph/tool_requires/using_protobuf/myserver/test_package/CMakeLists.txt
Comment (memsharded) [inline examples/graph/tool_requires/using_protobuf/myserver/test_package/CMakeLists.txt:4]: protobuf not necessary as a direct dependency? Typically only ``myserver`` needs a ``find_package()``

File: examples/graph/tool_requires/using_protobuf/myserver/src/myserver.cpp
Comment (memsharded) [inline examples/graph/tool_requires/using_protobuf/myserver/src/myserver.cpp:6]: no actual usage of protobuf?

File: examples/graph/tool_requires/using_protobuf/myserver/src/myserver.cpp
Comment (memsharded) [inline examples/graph/tool_requires/using_protobuf/myserver/src/myserver.cpp:25]: All this extra output can be removed for simplicity

File: examples/graph/tool_requires/using_protobuf/myserver/test_package/src/example.cpp
Comment (memsharded) [inline examples/graph/tool_requires/using_protobuf/myserver/test_package/src/example.cpp:7]: ``test_package`` of ``myserver`` doesnt' test ``myserver``?

**Review by memsharded — APPROVED** [review]

File: examples/graph/tool_requires/different_versions/run_example.py
Comment (franramirez688) [inline examples/graph/tool_requires/different_versions/run_example.py:11]:
```suggestion
    assert "MYGCC=2.0!!" in output

```

File: examples/graph/tool_requires/using_protobuf/myaddresser/apple-arch-armv8
Comment (czoido) [inline examples/graph/tool_requires/using_protobuf/myaddresser/apple-arch-armv8:7]:
```suggestion
compiler.version=13
```

File: examples/graph/tool_requires/using_protobuf/myaddresser/apple-arch-x86_64
Comment (czoido) [inline examples/graph/tool_requires/using_protobuf/myaddresser/apple-arch-x86_64:7]:
```suggestion
compiler.version=13
```

**Review by czoido — APPROVED** [review]

File: examples/graph/tool_requires/using_protobuf/run_example.py
Comment (czoido) [inline examples/graph/tool_requires/using_protobuf/run_example.py:21]:
```suggestion
output = run("file myaddresser/test_package/build/apple-clang-13-armv8-gnu17-release/example")
assert "64-bit executable arm64" in output
output = run("file myaddresser/test_package/build/apple-clang-13-x86_64-gnu17-release/example")
assert "64-bit executable x86_64" in output
```

**Pull request merged**
